### PR TITLE
added component for stjude.

### PR DESCRIPTION
**Pull request by SunburnedGoose**
Created a component to advertise the donation link, with a small blurb to support why we want to help.

I've added padding to the bottom of the inventory view to avoid overlap of the ad with the items. It renders below Loadouts and Comparison.

### CODE REVIEW

**Comment by SunburnedGoose** [comment]
Screen Shot 2019-05-28 at 12 40 14 AM
Screen Shot 2019-05-28 at 12 40 25 AM
Screen Shot 2019-05-28 at 12 40 39 AM

File: src/app/inventory/Inventory.tsx [hunk 1]
```diff
@@ -69,6 +72,7 @@ class Inventory extends React.Component<Props> {
     return (
       <>
         <Stores />
+        {!mobile && <StJudeToaster />}
```
Comment (bhollis) [inline src/app/inventory/Inventory.tsx]: We also have an `isPhonePortrait` in redux that can be used to detect phone mode.

File: src/app/stJude/StJudeToaster.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import classNames from 'classnames';
+import { AppIcon, disabledIcon } from '../shell/icons';
+import './st-jude-toaster.scss';
+import stJudeImage from '../../images/stjude_finding-cures_londyn-learn-more_animated_300x50.gif';
+
+const StJudeToaster = () => {
+  const [isMinimized, setIsMinimized] = useState(true);
+  const [visible, setVisible] = useState(false);
+
+  useEffect(() => {});
```
Comment (bhollis) [inline src/app/stJude/StJudeToaster.tsx]: Remove this?

File: src/app/stJude/StJudeToaster.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  return (
+    <>
+      <div
+        className={classNames(`st-jude-toaster`, {
+          minimized: isMinimized,
+          maximized: !isMinimized
+        })}
+      >
+        {!isMinimized && (
```
Comment (bhollis) [inline src/app/stJude/StJudeToaster.tsx]: You can write this as a ternary, like `isMinimized ? (some content) : (other content)`.

File: src/app/stJude/StJudeToaster.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          <div className={classNames(`st-jude-info`, { visible })}>
+            <span className="st-jude-hide" onClick={closeCampaign}>
+              <span className="sheet-close">
```
Comment (bhollis) [inline src/app/stJude/StJudeToaster.tsx]: Any reason to not use the actual Sheet component? It'll inherit swipe-to-close and other stuff.

File: src/app/stJude/StJudeToaster.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                <AppIcon icon={disabledIcon} />
+              </span>
+            </span>
+            <p>
+              The team at DIM joins the Destiny community to urge you to support St. Jude Children's
+              Research Hospital. Your support helps ensures that families will never receive a bill
+              from St. Jude for treatment, travel, housing or food.
```
Comment (bhollis) [inline src/app/stJude/StJudeToaster.tsx]: Mind throwing this in a translation?

File: src/app/stJude/st-jude-toaster.scss [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+.store-buckets {
+  margin-bottom: 60px;
+}
+
+.android .store-buckets,
+.ios .store-buckets {
+  margin-bottom: 0;
+}
```
Comment (bhollis) [inline src/app/stJude/st-jude-toaster.scss]: Might be copy/paste, I don't think you need any store-buckets stuff.

Comment (SunburnedGoose) [reply src/app/stJude/st-jude-toaster.scss]: I need to add the margin on the buckets so the ad doesn't block access to items, and then zero out that margin if the user is on mobile.  An alternative is to wedge a component in the footer that grows and shrinks in the inventory view.

File: src/app/stJude/StJudeToaster.tsx (hunk 4 above)
Comment (SunburnedGoose) [reply src/app/stJude/StJudeToaster.tsx]: It was the transition between the text and the ad, and also defining the width.  I didn't want to hack into it for something that was short lived and desktop only. I can revisit.

**Pull request merged**